core/state/statedb.go

```diff
@@ -1372,6 +1379,15 @@ func (s *StateDB) Prepare(rules params.Rules, sender, coinbase common.Address, d
 	s.transientStorage = newTransientStorage()
 }
 
+// PrepareBlock prepares the statedb for execution of a block. It tracks
+// the addresses of enabled precompiles for debugging purposes.
+func (s *StateDB) PrepareBlock(precompiles []common.Address) {
+	s.precompiles = make(map[common.Address]struct{})
```

Comment:
We should probably pre-allocate:

```
s.precompiles = make(map[common.Address]struct{}, len(precompiles))
```

Reply:
Aha this line can actually be omitted. I allocate the map during statedb instantiation. I prefer to do it there not to hit a nil reference panic by accident.